File: nmLabReport/lab23.c
#include <stdio.h>

// Function to be integrated
double f(double x) {
    return x * x; // Example function: f(x) = x^2
}

// Composite Trapezoidal Rule function
double trapezoidal(double a, double b, int n) {
    double h = (b - a) / n;
    double sum = 0.5 * (f(a) + f(b));
    
    for (int i = 1; i < n; i++) {
        sum += f(a + i * h);
    }
    
    return sum * h;
}

int main() {
    double a, b;
    int n;
    
    // Input the limits of integration and number of subintervals
    printf("Enter the lower limit a: ");
    scanf("%lf", &a);
    printf("Enter the upper limit b: ");
    scanf("%lf", &b);
    printf("Enter the number of subintervals n: ");
    scanf("%d", &n);
    
    // Calculate the integral
    double result = trapezoidal(a, b, n);
    
    // Output the result
    printf("The integral is: %lf\n", result);
    
    return 0;
}

File: nmLabReport/lab23.py
def f(x):
    return x * x  # Example function: f(x) = x^2

def trapezoidal(a, b, n):
    h = (b - a) / n
    sum = 0.5 * (f(a) + f(b))
    
    for i in range(1, n):
        sum += f(a + i * h)
    
    return sum * h

def main():
    a = float(input("Enter the lower limit a: "))
    b = float(input("Enter the upper limit b: "))
    n = int(input("Enter the number of subintervals n: "))
    
    result = trapezoidal(a, b, n)
    print(f"The integral is: {result}")

if __name__ == "__main__":
    main()